quickevent/app/quickevent/plugins/Event/src/services/oresultsclient.cpp

```diff
@@ -143,4 +147,173 @@ void OResultsClient::sendFile(QString name, QString request_path, QString file)
 	});
 }
 
+void OResultsClient::onDbEventNotify(const QString &domain, int connection_id, const QVariant &data)
+{
+	if (status() != Status::Running)
+		return;
+	Q_UNUSED(connection_id)
+	if(domain == QLatin1String(Event::EventPlugin::DBEVENT_CARD_PROCESSED_AND_ASSIGNED)) {
+		auto checked_card = quickevent::core::si::CheckedCard(data.toMap());
+		int competitor_id = getPlugin<RunsPlugin>()->competitorForRun(checked_card.runId());
+		onCompetitorChanged(competitor_id);
+	}
+	if(domain == QLatin1String(Event::EventPlugin::DBEVENT_COMPETITOR_EDITED)) {
+		int competitor_id = data.toInt();
+		onCompetitorChanged(competitor_id);
+	}
+}
+
+void OResultsClient::sendCompetitorChange(QString xml) {
+	QUrl url(API_URL + "/meos");
+	QNetworkRequest request(url);
+	request.setRawHeader("pwd", settings().apiKey().toUtf8());
+	request.setHeader( QNetworkRequest::ContentTypeHeader, "application/xml" );
+	QNetworkReply *reply = m_networkManager->post(request, xml.toUtf8());
+
+	connect(reply, &QNetworkReply::finished, reply, [reply]()
+	{
+		if(reply->error()) {
+			qfError() << "OReuslts.eu [competitor change]:" << reply->errorString();
+		}
+		else {
+			QString msg = reply->readAll();
+			if (msg.contains("BADPWD"))
+				qfError() << "OReuslts.eu [competitor change]: Invalid API key";
+			else if (!msg.contains("OK"))
+				qfError() << "OReuslts.eu [competitor change]: Failed to process request";
+		}
+		reply->deleteLater();
+	});
+}
+
+static void append_list(QVariantList &lst, const QVariantList &new_lst)
+{
+	lst.insert(lst.count(), new_lst);
+}
+
+static bool is_csos_reg(QString &reg)
+{
+	const std::regex csos_registration_regex("[A-Z]{3}[0-9]{4}");
```

Comment:
what about static?

Reply:
yep, it needs to be there :upside_down_face: